src/PrestaShopBundle/Translation/PrestaShopTranslatorTrait.php

```diff
@@ -209,4 +201,23 @@ private function normalizeDomain($domain)
 
         return $normalizedDomain;
     }
+
+    /**
+     * Replaces special characters on the message
+     *
+     * @param string $message the message
+     * @param callable|string $functionName Name of function to be called
+     *
+     * @return string
+     */
+    private function replaceSpecialCharsWithLegacyFunctions(string $message, string|callable $functionName): string
```

Comment:
Union types is a PHP 8 feature, we can't use it yet if we want to be compatible with 7.2+
```suggestion
    private function replaceSpecialCharsWithLegacyFunctions(string $message, string $functionName): string
```